### PR TITLE
Add new params to SetASSVirtualDevice for snapshot mgr passwd and admin ...

### PR DESCRIPTION
**Pull request by rajsing**
...passwd

### CODE REVIEW

**Comment by rajsing** [comment]
Have also clubbed the changes in param orders to bring mandatory ones before the optional ones in this PR

File: src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs [hunk 1]
```diff
@@ -518,5 +518,54 @@ internal bool IsValidAsciiString(string s)
...
+            foreach(var regexStr in criteriaRegexs){
+                var regex = new Regex(regexStr);
+                if(regex.IsMatch(data)){
+                    criteriaFulfilled += 1;
+                }
+            }
+
+            // If atleast 3 criteria have been fulfilled, then the password is complex enough
+            if(criteriaFulfilled < 3){
+                throw new ArgumentException(errorMessage);
```
Comment (parvezah) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: This error message seems to be generic and wont specify which password didn't meet the criteria. Can you also add  which password didnt meet the criteria.

File: src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs [hunk 2]
```diff
@@ -518,5 +518,54 @@ internal bool IsValidAsciiString(string s)
...
+        internal void ValidatePasswordComplexity(string data)
+        {
+            string errorMessage = Resources.PasswordCharacterCriteriaError;
+            var criteriaFulfilled = 0;
+            // Regular expressions for lowercase letter, uppercase letter, digit and special char
+            // respectively
+            string[] criteriaRegexs = { ".*[a-z]", ".*[A-Z]", ".*\\d", ".*\\W" };
+
+            foreach(var regexStr in criteriaRegexs){
+                var regex = new Regex(regexStr);
```
Comment (Sumanth-SK) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: Please instantiate a regex object once and reuse it, as regex object create is a costly operation

File: src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs (hunk 1 above)
Comment (rajsing) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: I have fixed this somehow the preview here is not getting updated you can check the diff

File: src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs (hunk 2 above)
Comment (rajsing) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: fixed.

**Pull request merged**